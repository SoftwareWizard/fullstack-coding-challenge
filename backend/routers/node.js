const express = require('express');
const router = express.Router();

const node = require('../data/node.data.json');

router.get('/', (request, response) => {
    return response.send(node);
});

module.exports = router;