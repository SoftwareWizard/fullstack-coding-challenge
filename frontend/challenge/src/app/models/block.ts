export interface IBlock {
  id: string;
  blockNumber: number;
  nodeId: string;
  description: string;
}
