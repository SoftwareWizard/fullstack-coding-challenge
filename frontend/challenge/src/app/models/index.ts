export * from './block-attributes';
export * from './block';
export * from './node';
export * from './status';
