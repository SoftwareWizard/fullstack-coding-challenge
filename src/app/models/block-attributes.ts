export interface BlockAttributes {
  index: number;
  timestamp: number;
  data: string;
  'previous-hash': string;
  hash: string;
}
