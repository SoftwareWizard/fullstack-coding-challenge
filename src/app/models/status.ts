export interface Status {
  node_name: string;
}
