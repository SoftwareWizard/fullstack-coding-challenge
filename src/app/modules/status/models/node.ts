export interface Node {
  url: string;
  name: string;
  status?: string;
  isLoading: boolean;
  isOnline: boolean;
  isExpanded: boolean;
}
