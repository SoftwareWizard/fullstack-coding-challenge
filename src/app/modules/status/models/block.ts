import { BlockAttributes } from './block-attributes';

export interface Block {
  id: string;
  type: string;
  attributes: BlockAttributes;
}
