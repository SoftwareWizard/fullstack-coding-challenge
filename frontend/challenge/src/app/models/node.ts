import { Status } from './status';

export interface INode {
  id: string;
  name: string;
  url: string;
  state: Status;
}
