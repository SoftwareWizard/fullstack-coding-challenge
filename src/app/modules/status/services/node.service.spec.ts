import { createHttpFactory, HttpMethod, SpectatorHttp } from '@ngneat/spectator/jest';

import { Block } from '../models';
import { BlockAttributes } from '../models/block-attributes';
import { API_SEGMENT, NodeService } from './node.service';

describe('NodeService', () => {
  let spectator: SpectatorHttp<NodeService>;
  let nodeService: NodeService;

  const TEST_URL = 'http://testurl.com';

  const createHttp = createHttpFactory(NodeService);

  beforeEach(() => {
    spectator = createHttp();
    nodeService = spectator.service;
  });

  it('should return node', () => {
    // act
    nodeService.getStatus(TEST_URL).subscribe();

    // assert
    spectator.expectOne(`${TEST_URL}/${API_SEGMENT}/status`, HttpMethod.GET);
  });

  it('should return blocks', async () => {
    // arrange
    const testBlock: Block = {
      id: 'testId',
      type: 'block',
      attributes: {} as BlockAttributes
    };

    const testBlockData = {
      data: [testBlock]
    };

    // act
    const promise = nodeService.getBlocks(TEST_URL);

    // assert
    spectator.expectOne(`${TEST_URL}/${API_SEGMENT}/blocks`, HttpMethod.GET).flush(testBlockData);

    const result = await promise;
    expect(result).toEqual(testBlockData.data);
  });
});
