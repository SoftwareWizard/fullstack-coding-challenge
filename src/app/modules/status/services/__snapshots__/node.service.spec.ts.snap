// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeService should return blocks 1`] = `
Array [
  Object {
    "attributes": Object {},
    "id": "testId",
    "type": "block",
  },
]
`;
